src/main/java/org/opensearch/sdk/BaseExtension.java

```diff
@@ -35,10 +36,40 @@ public abstract class BaseExtension implements Extension {
     protected ThreadPool threadPool;
 
     /**
-     * Empty constructor to fulfill abstract class requirements
+     * Optional classpath-relative path to a yml file containing extension settings.
+     */
+    protected static final String EXTENSION_SETTINGS_PATH = "/sample/helloworld-settings.yml";
```

Comment:
This path constant (if used) needs to stay in the Extension subclass/implementation.  This `BaseExtension` class shouldn't have any extension-specific code.